cmd/ci-secret-bootstrap/main.go

```diff
@@ -457,6 +469,132 @@ func writeSecrets(secretsMap map[string][]*coreapi.Secret, w io.Writer) error {
 	return nil
 }
 
+type comparable struct {
+	fields      sets.String
+	attachments sets.String
+	hasPassword bool
+}
+
+func constructBWItemsByName(items []bitwarden.Item) map[string]*comparable {
+	bwComparableItemsByName := make(map[string]*comparable)
+	for _, item := range items {
+		comparableItem := &comparable{}
+
+		if len(item.Fields) > 0 {
+			fields := sets.NewString()
+			for _, item := range item.Fields {
+				fields.Insert(item.Name)
+			}
+
+			comparableItem.fields = fields
+		}
+
+		if len(item.Attachments) > 0 {
+			attachments := sets.NewString()
+			for _, attachment := range item.Attachments {
+				attachments.Insert(attachment.FileName)
+			}
+
+			comparableItem.attachments = attachments
+		}
+
+		if item.Login != nil {
+			comparableItem.hasPassword = true
+		}
+
+		bwComparableItemsByName[item.Name] = comparableItem
+	}
+
+	return bwComparableItemsByName
+}
+
+func constructConfigItemsByName(config secretbootstrap.Config) map[string]*comparable {
+	cfgComparableItemsByName := make(map[string]*comparable)
+
+	for _, cfg := range config.Secrets {
+		for _, bwContext := range cfg.From {
+			if bwContext.BWItem != "" {
+				item, ok := cfgComparableItemsByName[bwContext.BWItem]
+				if !ok {
+					item = &comparable{
+						fields:      sets.NewString(),
+						attachments: sets.NewString(),
+					}
+				}
+				item.attachments.Insert(bwContext.Attachment)
+				item.fields.Insert(bwContext.Field)
+
+				if bwContext.Attribute == secretbootstrap.AttributeTypePassword {
+					item.hasPassword = true
+				}
+				cfgComparableItemsByName[bwContext.BWItem] = item
+			}
+
+			if len(bwContext.DockerConfigJSONData) > 0 {
+				for _, context := range bwContext.DockerConfigJSONData {
+					item, ok := cfgComparableItemsByName[context.BWItem]
+					if !ok {
+						item = &comparable{
+							fields:      sets.NewString(),
+							attachments: sets.NewString(),
+						}
+					}
+					item.fields.Insert(context.RegistryURLBitwardenField)
```

Comment:
I think we need to check if this is an empty string first

Reply:
I think that sets library won't insert anything if the string is empty. I will double check though.